### PR TITLE
Penalties when fighting while prone/crouching

### LINKED ISSUE #51735: You are completely unable to fight in melee at all with a broken leg
### Describe the bug

If you break a leg, you immediately fall over into prone whenever you try to walk or stand back up. While in this state, you are given an error: "You cannot fight while on the ground" that prevents you from attacking no matter what. No matter what I tried to do (moving after falling over, standing back up and immediately attacking, passing turns) I was unable to attack whatsoever. Going prone manually, without broken legs, does _NOT_ cause this issue.

### Steps To Reproduce

1. Spawn a character in, debug spawn in a test dummy or zombie.
2. Debug damage your leg to broken.
3. Observe you are entirely unable to attack, even in a debuffed state, leaving you 100% helpless unless you have a firearm.

### Expected behavior

Obviously, fighting while prone should be difficult, especially if you're also dealing with a broken leg. But if a zombie (or in my case, a mutant coyote) is pouncing on you while you're lying there, you should still be able to feebly fight back in desperation rather than just lying there like a lemon. I'm not sure if there necessarily needs to be any special debuffs for this beyond the normal fight debuffs for having a broken leg, the pain from it breaking, and being prone. So, to clarify, I obviously don't think you should be able to fight unimpeded, and naturally you shouldn't be able to chase anything down (though this is already handled by prone's massive movecost) or dodge, but you should still be able to swipe your combat knife or flail somewhat ineffectually with a longsword if an enemy is actively dogpiling you.

### Screenshots

image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 09095af [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dseguin**
#### Summary
None

#### Purpose of change
Incidentally fixes #51735 by allowing attacks while prone with a broken leg.

Attacking while prone has the same outcome as attacking while standing. These changes add some penalties to make the player's stance more important in fights (other than move cost).

#### Describe the solution
The following penalties were added:

Accuracy:
- *When prone:* Large penalty (-8 to hit)
- *When crouching:* Small penalty (-2 to hit)

Damage:
- *When prone:* Reduced to 30% of normal damage
- *When crouching:* Reduced to 80% of normal damage

Stumble:
- *When prone:* Reduce STR to 25% for stumble calculation
- *When crouching:* Reduce STR to 50% for stumble calculation

Attack speed:
- *When prone:* 4x movement cost penalty
- *When crouching:* 1.5x movement cost penalty

Stamina cost:
- *When prone:* Increase stamina usage by 50 points
- *When crouching:* Increase stamina usage by 20 points

Also, added a message to `Character::get_miss_reason()` when attacking while prone.
combat_msg

#### Testing
1. Spawn a test dummy
2. Reduce leg health to 0
3. Attempt to hit dummy while prone -> character attacks
4. Spawn a zombie
5. Attempt to hit zombie while prone -> lower hit rate, less damage
6. Same for crouching

#### Additional context
Some more testing would probably be needed to tweak the values to be more realistic.

### CODE REVIEW

**Comment by actual-nh** [comment]
Some martial arts (e.g., wrestling) should at least partially be able to negate some of these penalties. Ping: @Hymore246?

**Comment by Hymore246** [comment]
> 
> 
> Some martial arts (e.g., wrestling) should at least partially be able to negate some of these penalties. Ping: @Hymore246?

Kinda. The thing to remember about most ground fighting martial arts is that they use a different set of attacks when on the ground compared to standing. Furthermore, a wrestler laying on the ground is going to have problems attacking a standing enemy. You usually want to match the position of someone you are fighting. I could see some styles reducing some of the penalties for crouching but due to the way the martial arts system works, you should never crouch or go prone willingly. And if you are forced into doing it, you are probably close to death with massive penalties already.

**Comment by actual-nh** [comment]
> Kinda. The thing to remember about most ground fighting martial arts is that they use a different set of attacks when on the ground compared to standing. Furthermore, a wrestler laying on the ground is going to have problems attacking a standing enemy. You usually want to match the position of someone you are fighting.

Sacrifice takedown? In other words, there are a number of attacks (tackles are another one) in which yes, you wind up down, but so does the opponent. (Admittedly mainly useful against a single opponent!)

>I could see some styles reducing some of the penalties for crouching but due to the way the martial arts system works, you should never crouch or go prone willingly.

Even if the opponent has been forced to do so also? (I don't know whether some low stances should be considered crouching or not.)

**Comment by Hymore246** [comment]
> Sacrifice takedown? In other words, there are a number of attacks (tackles are another one) in which yes, you wind up down, but so does the opponent. (Admittedly mainly useful against a single opponent!)
 
The randomness of the martial art systems means that grapples and sacrifice takedowns are REALLY bad idea because you cannot control them. You WILL tackle a zombie into a horde and get yourself killed. This is why I avoided using them in the Judo rebalance.

> Even if the opponent has been forced to do so also? (I don't know whether some low stances should be considered crouching or not.)

You can down an opponent but they cannot crouch as far as I know. You would be at an advantage over a crouching/downed opponent if you are standing. At best, I would say you could break even with them in the sense that you would both have penalties that cancel out.

File: src/melee.cpp [hunk 1]
```diff
@@ -322,6 +322,13 @@ float Character::hit_roll() const
         hit -= 2.0f;
     }
 
+    // Difficult to land a hit while prone
+    if( is_on_ground() ) {
+        hit -= one_in( 4 ) ? 4.0f : 10.0f;
+    } else if( is_crouching() ) {
+        hit -= one_in( 4 ) ? 0.0f : 4.0f;
+    }
```
Comment (kevingranade) [inline src/melee.cpp:330]: I don't like injecting randomness in a bunch of different places as it makes the final outcome difficult to predict, make this a flat penalty.

Comment (dseguin) [reply src/melee.cpp]: I changed these to flat values (8.0f and 2.0f respectively).

**Pull request merged**